src/tokens.rs

```diff
@@ -219,16 +222,53 @@ pub async fn get_token_symbol_by_mint(mint: &str) -> anyhow::Result<String> {
     Ok(response.symbol)
 }
 
+/// Get token symbol from Metaplex Fungible Token Metadata
+/// https://docs.metaplex.com/programs/token-metadata/accounts#metadata
+/// Recommended method since 2022-06
+pub async fn get_token_symbol_by_mint_from_metadata(client: &RpcClient, mint: &Pubkey) -> anyhow::Result<String> {
+    let metadata_program_id = solana_sdk::pubkey!("metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s");
+
+    let (metadata_address, _) = Pubkey::find_program_address(
+        &[b"metadata", metadata_program_id.as_ref(), mint.as_ref()],
+        &metadata_program_id,
+    );
+    let metadata = client.get_account_data(&metadata_address).await?;
+
+    // The on-chain symbol of the token, limited to 10 bytes
+    // Offset - 101, size 14
+    let symbol = String::try_from_slice(&metadata[101..115])?;
+
+    Ok(symbol.trim_end_matches('\0').to_owned())
```

Comment:
I used `trim_end_matches` because I didn't use `borsh` to deserialize. Now it shouldn't be needed.

Reply:
Tests don't pass without it. In spite of the size in borsh spec, the field filled with zeroes which have to be trimmed.